docs/spec/spec.md

```diff
@@ -0,0 +1,283 @@
+    string name;
+    # The name of the database
+    string database;
+    # The name or ARN of the secret that enables access to the database
+    string secretArn?;
+    # The number of seconds to keep the session alive after the query finishes
+    int sessionKeepAliveSeconds?;
+|};
+```
+
+### 2.2. Initialization
+
+- A `redshiftdata:Client` can be initialized by providing the `redshiftdata:ConnectionConfig`.
+
+```ballerina
+# Initialize AWS Redshift Data API client.
+# ```
+# redshiftdata:Client redshiftdata = check new (region = redshiftdata:US_EAST_2,
+#    authConfig = {
+#        accessKeyId: "<aws-access-key>",
+#        secretAccessKey: "<aws-secret-key>"
+#    },
+#    dbAccessConfig = {
+#        id: "<cluster-id>",
+#        database: "<database-name>",
+#        dbUser: "<db-user>"
+#    }
+# );
+#
+# + connectionConfig - The Redshift Data API client configurations
+# If a `dbAccessConfig` is provided, it will be used for the statement executions and it can be overridden 
+# using the `dbAccessConfig` at the API level
+# + return - The `redshiftdata:Client` or a `redshiftdata:Error` if the initialization fails
+public isolated function init(*redshift:ConnectionConfig connectionConfig) returns redshift:Error?;
+```
+
+### 2.3. Execution configurations
+
+- When executing a SQL statement on AWS redshift, the developer could provide additiona configurations which 
+is related to the query execution. The `ExecutionConfig` record represents the additional configuration which 
+could be used when executing a single SQL statement or a SQL statement batch.
+
+```ballerina
+public type ExecutionConfig record {|
+    # The database access configurations for the Redshift Data
+    Cluster|WorkGroup|SessionId dbAccessConfig?;
+    # A unique, case-sensitive identifier that you provide to ensure the idempotency of the request
+    string clientToken?;
+    # The name of the SQL statement
+    string statementName?;
+    # Flag which indicates to send an event after the SQL statement execution to 
+    # an event bus instance running in Amazon EventBridge
+    boolean withEvent?;
+|};
+```
+
+### 2.4. Functions
+
+- To run a SQL statement on AWS Redshift instance, `execute` function can be used.
+
+```ballerina
+# Runs an SQL statement, which can be data manipulation language (DML) or data definition language (DDL).
+# ```
+# redshiftdata:ExecutionResponse response = check redshiftdata->execute(`SELECT * FROM Users`);
+# ```
+#
+# + statement - The SQL statement to be executed
+# + executionConfig - The configurations related to the execution of the statement
+# + return - The `redshiftdata:ExecutionResponse` or a `redshiftdata:Error` if the execution fails
+remote isolated function execute(sql:ParameterizedQuery statement, *redshift:ExecutionConfig executionConfig) returns redshift:ExecutionResponse|redshift:Error;
+```
+
+- To run multiple SQL statements on AWS Redshift instance, `batchExecute` function can be used.
```

Comment:
Shall we explain a bit about the behaviour or calling execute and batchExecute functions?
Like how to execute, describe, and get result?
Maybe an example would be good.

Reply:
Added a separate section to include example usage with following commits:
- [d08f72329fd6d3ef9614ffbc74f491e99af3c71d](https://github.com/ballerina-platform/module-ballerinax-aws.redshiftdata/pull/26/commits/d08f72329fd6d3ef9614ffbc74f491e99af3c71d)
- [8891b7b39cbc7161ca3a2337bafce46760ad4ee8](https://github.com/ballerina-platform/module-ballerinax-aws.redshiftdata/pull/26/commits/8891b7b39cbc7161ca3a2337bafce46760ad4ee8)